include/cuco/static_multimap.cuh

```diff
@@ -0,0 +1,1808 @@
+     * the empty value sentinels into the output only if `is_outer` is true.
+     *
+     * @tparam cg_size The number of threads in CUDA Cooperative Groups
+     * @tparam buffer_size Size of the output buffer
+     * @tparam is_outer Boolean flag indicating whether outer join is peformed
+     * @tparam CG Cooperative Group type
+     * @tparam atomicT Type of atomic storage
+     * @tparam OutputZipIt1 Device accessible output zip iterator for probe matches
+     * @tparam OutputZipIt2 Device accessible output zip iterator for contained matches
+     * @tparam PairEqual Binary callable type
+     * @param g The Cooperative Group used to retrieve
+     * @param pair The pair to search for
+     * @param cg_counter Pointer to the CG counter
+     * @param probe_output_buffer Buffer of the matched probe pair sequence
+     * @param contained_output_buffer Buffer of the matched contained pair sequence
+     * @param num_matches Size of the output sequence
+     * @param probe_output_begin Beginning of the output sequence of the matched probe pairs
+     * @param contained_output_begin Beginning of the output sequence of the matched contained
+     * pairs
+     * @param pair_equal The binary callable used to compare two pairs for equality
+     */
+    template <uint32_t cg_size,
+              uint32_t buffer_size,
+              bool is_outer,
+              typename CG,
+              typename atomicT,
+              typename OutputZipIt1,
+              typename OutputZipIt2,
+              typename PairEqual>
+    __device__ void pair_retrieve_impl(CG const& g,
+                                       value_type const& pair,
+                                       uint32_t* cg_counter,
+                                       value_type* probe_output_buffer,
+                                       value_type* contained_output_buffer,
+                                       atomicT* num_matches,
+                                       OutputZipIt1 probe_output_begin,
+                                       OutputZipIt2 contained_output_begin,
+                                       PairEqual pair_equal) noexcept;
+
+   public:
+    /**
+     * @brief Flushes per-CG buffer into the output sequence.
+     *
+     * @tparam CG Cooperative Group type
+     * @tparam atomicT Type of atomic storage
+     * @tparam OutputIt Device accessible output iterator whose `value_type` is
+     * convertible to the map's `mapped_type`
+     * @param g The Cooperative Group used to flush output buffer
+     * @param num_outputs Number of valid output in the buffer
+     * @param output_buffer Buffer of the key/value pair sequence
+     * @param num_matches Size of the output sequence
+     * @param output_begin Beginning of the output sequence of key/value pairs
+     */
+    template <typename CG, typename atomicT, typename OutputIt>
+    __inline__ __device__ void flush_output_buffer(CG const& g,
```

Comment:
I think this function needs more documentation about when/how it should be used.